scripts/globals/battlefield.lua

```diff
@@ -68,10 +70,788 @@ xi.battlefield.dropChance =
+        })
+        self.hasListeners = true
+    end
+
+    if setupEntryNpc and self.entryNpc then
+        utils.append(zoneSection, {
+            [self.entryNpc] =
+            {
+                onTrade = Battlefield.onEntryTrade,
+                onTrigger = Battlefield.onEntryTrigger,
+            }
+        })
+    end
+
+    if setupExitNpc and self.exitNpc then
+        utils.append(zoneSection, {
+            [self.exitNpc] =
+            {
+                onTrigger = Battlefield.onExitTrigger,
+            }
+        })
+    end
+
+    xi.battlefield.contents[self.battlefieldId] = self
+    if utils.hasKey(self.zoneId, xi.battlefield.contentsByZone) then
+        table.insert(xi.battlefield.contentsByZone[self.zoneId], self)
+    else
+        xi.battlefield.contentsByZone[self.zoneId] = { self }
+    end
+    return self
+end
+
+function Battlefield:checkRequirements(player, npc, registrant, trade)
+    if (self.entryNpc ~= npc:getName()) then
+        return false
+    end
+
+    for _, keyItem in ipairs(self.requiredKeyItems) do
+        if not player:hasKeyItem(keyItem) then
+            return false
+        end
+    end
+
+    if trade and #self.requiredItems > 0 then
+        if not npcUtil.tradeHasExactly(trade, self.requiredItems) then
+            return false
+        end
+    end
+
+    return true
+end
+
+function Battlefield:checkSkipCutscene(player)
+    return false
+end
+
+function Battlefield.onEntryTrade(player, npc, trade, onUpdate)
+    -- Check if player's party has level sync
+    if xi.battlefield.rejectLevelSyncedParty(player, npc) then
+        return false
+    end
+
+    -- Validate trade
+    if not trade then
+        return false
+    end
+
+    -- Validate battlefield status
+    if player:hasStatusEffect(xi.effect.BATTLEFIELD) and not onUpdate then
+        player:messageBasic(xi.msg.basic.WAIT_LONGER, 0, 0)
+        return false
+    end
+
+    -- Check if another party member has battlefield status effect. If so, don't allow trade.
+    local alliance = player:getAlliance()
+    for _, member in pairs(alliance) do
+        if member:hasStatusEffect(xi.effect.BATTLEFIELD) then
+            player:messageBasic(xi.msg.basic.WAIT_LONGER, 0, 0)
+
+            return false
+        end
+    end
+
+    local zoneId = player:getZoneID()
+
+    -- Determine which battlefields are available given the traded items
+    local options = xi.battlefield.getBattlefieldOptions(player, npc, trade)
+    if options == 0 then
+        local noEntryMessage = zones[zoneId].text.NO_BATTLEFIELD_ENTRY
+        if noEntryMessage then
+            player:messageSpecial(noEntryMessage)
+        end
+        return false
+    end
+
+    -- Ensure that the traded item(s) are not worn out
+    local contents = xi.battlefield.contentsByZone[zoneId]
+    for _, content in ipairs(contents) do
+        if #content.requiredItems > 0 and content.requiredItems[1].wornMessage and player:hasWornItem(content.requiredItems[1]) then
```

Comment:
This could probably be easier to read if broken out into multiline conditional style:
```
if
    condition1 and
    not condition2
then

end
```